### PR TITLE
Fix fatal error with using the -b option.

### PR DESCRIPTION
**Pull request by arizvisa**
This fixes Issue #1168.

### CODE REVIEW

File: bootstrap-salt.sh [hunk 1]
```diff
@@ -6576,7 +6576,7 @@ if [ "$_CUSTOM_MASTER_CONFIG" != "null" ] || [ "$_CUSTOM_MINION_CONFIG" != "null
         _TEMP_CONFIG_DIR="$_SALT_ETC_DIR"
     fi
 
-    if [ "$_CONFIG_ONLY" -eq $BS_TRUE ]; then
+    if [ ${NO_DEPS} -eq $BS_FALSE -a "$_CONFIG_ONLY" -eq $BS_TRUE ]; then
```
Comment (rallytime) [inline bootstrap-salt.sh]: `${NO_DEPS}` should be `${_NO_DEPS}` here, yes?

**Comment by arizvisa** [comment]
yep, my bad. you're right. the variable should've been _NO_DEPS.

File: bootstrap-salt.sh (hunk 1 above)
Comment (arizvisa) [reply bootstrap-salt.sh]: yep, my bad. you're right. the variable should've been _NO_DEPS.

**Comment by rallytime** [comment]
@arizvisa Ok, now we have some lint errors to take care of. :)

https://jenkins-testing.saltstack.com/job/bootstrap/job/pr/job/1169/job/lint/5/violations/file/bootstrap-salt.sh/

Can you fix those up, and then we can get this in. Basically, we need to be using `&&` instead of `-a`.

**Comment by arizvisa** [comment]
ah well, i don't care enough to opensource.

it's a shellscript. manpage for '[' and 'test' both say -a is for "and". if you want to fix -b, you know what to do. enjoy!

**Comment by vutny** [comment]
@arizvisa Thanks for the fix, anyhow.

By the way, there's a sense behind having `&&` everywhere. Both `-a` and `-o` to mean AND and OR in a [ .. ] test expression are not well defined. They are obsolescent extensions in POSIX and the lack of additional quoting levels means that expressions involving these operators are often ambiguous. Values that begin with hyphens and the `!` string could become issues.

**Pull request merged**